### PR TITLE
Move datafmt docs out, update them a bit

### PR DESCRIPTION
**Pull request by kshyatt**
These docstrings could stand some real improvement so if someone has suggestions, I'm happy to update them (or you can just push to my branch).

### CODE REVIEW

File: doc/stdlib/io-network.rst [hunk 1]
```diff
@@ -624,7 +624,7 @@ Text I/O
 
    The columns are assumed to be separated by one or more whitespaces. The end of line delimiter is taken as ``\n``\ . If all data is numeric, the result will be a numeric array. If some elements cannot be parsed as numbers, a heterogeneous array of numbers and strings is returned.
 
-.. function:: writedlm(f, A, delim='\\t')
+.. function:: writedlm(f, A, dl='\\t'; opts)
```
Comment (tkelman) [inline doc/stdlib/io-network.rst]: why delim everywhere but here?

Comment (kshyatt) [inline doc/stdlib/io-network.rst]: Because in the arg list it's `dl`. I can change it back :)

Comment (tkelman) [inline doc/stdlib/io-network.rst]: yeah I think doc consistency will be more visible than code naming consistency. we could change the latter too if we feel like it.

File: base/docs/helpdb/Base.jl [hunk 2]
```diff
@@ -10,17 +10,11 @@ Raises a `SystemError` for `errno` with the descriptive string `sysfunc` if `ift
...
-(either a filename string or an `IO` stream) using the given delimiter `delim` (which
-defaults to tab, but can be any printable Julia object, typically a `Char` or
-`AbstractString`).
-
-For example, two vectors `x` and `y` of the same length can be written as two columns of
-tab-delimited text to `f` by either `writedlm(f, [x y])` or by `writedlm(f, zip(x, y))`.
+Compute the digamma function of `x` (the logarithmic derivative of `gamma(x)`)
 """
-writedlm
+digamma
```
Comment (tkelman) [inline base/docs/helpdb/Base.jl]: you moved this docstring already, I think adding it back was a rebase mistake

Comment (kshyatt) [inline base/docs/helpdb/Base.jl]: I did aaaaa

File: doc/stdlib/io-network.rst (hunk 1 above)
Comment (tkelman) [inline doc/stdlib/io-network.rst]: bump

Comment (kshyatt) [inline doc/stdlib/io-network.rst]: ty for the reminder i will take care of it tonight :)

**Pull request merged**